### PR TITLE
JAMES-2586 - Introduce PostgresMailboxSessionMapperFactoryTODO & fixup bug

### PR DESCRIPTION
**Pull request by vttranlina**

### CODE REVIEW

**Review by Arsnael — COMMENTED** [review]
That's for fixing mpt imap tests, correct?

**Comment by vttranlina** [comment]
> That's for fixing mpt imap tests, correct?

yes.

**Review by Arsnael — APPROVED** [review]

File: mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/PostgresMailboxSessionMapperFactoryTODO.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    @Override
+    public SubscriptionMapper createSubscriptionMapper(MailboxSession session) {
+        return new PostgresSubscriptionMapper(new PostgresSubscriptionDAO(executorFactory.create(session.getUser().getDomainPart())));
+    }
+
+    /**
+     * Return a new {@link EntityManager} instance
+     *
+     * @return manager
+     */
```
Comment (chibenwa) [inline mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/PostgresMailboxSessionMapperFactoryTODO.java:108]:
```suggestion
```

File: mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/PostgresMailboxSessionMapperFactoryTODO.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    private EntityManager createEntityManager() {
+        return entityManagerFactory.createEntityManager();
```
Comment (chibenwa) [inline mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/PostgresMailboxSessionMapperFactoryTODO.java:110]: Needed for what?

File: mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/PostgresMailboxSessionMapperFactoryTODO.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    @Override
+    public PostgresModSeqProvider getModSeqProvider(MailboxSession session) {
+        return new PostgresModSeqProvider.Factory(executorFactory).create(session);
+    }
+
+    @Override
+    public AttachmentMapper createAttachmentMapper(MailboxSession session) {
+        return new JPAAttachmentMapper(entityManagerFactory);
```
Comment (chibenwa) [inline mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/PostgresMailboxSessionMapperFactoryTODO.java:130]: Discard this. It is not needed for IMAP. Return a fake one instead.

File: mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/PostgresMailboxSessionMapperFactoryTODO.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+import org.apache.james.mailbox.store.mail.AnnotationMapper;
+import org.apache.james.mailbox.store.mail.AttachmentMapper;
+import org.apache.james.mailbox.store.mail.AttachmentMapperFactory;
+import org.apache.james.mailbox.store.mail.MailboxMapper;
+import org.apache.james.mailbox.store.mail.MessageIdMapper;
+import org.apache.james.mailbox.store.mail.MessageMapper;
+import org.apache.james.mailbox.store.user.SubscriptionMapper;
+
+
+public class PostgresMailboxSessionMapperFactoryTODO extends MailboxSessionMapperFactory implements AttachmentMapperFactory {
```
[the remark this answers is not in the document]
Comment (vttranlina) [reply mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/PostgresMailboxSessionMapperFactoryTODO.java:49]: If renaming it now, the CI will be failed

[the remark this answers is not in the document]
Comment (chibenwa) [reply mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/PostgresMailboxSessionMapperFactoryTODO.java:49]: Curious why but OK

**Comment by Arsnael** [comment]
@vttranlina can you squash your fixup please?

**Pull request merged**